backend/xivdpscalc/character/__init__.py

```diff
@@ -24,96 +24,140 @@ class Character:
     The main class where damage is calculated. Initialized by providing a CharacterStatSpread.
     """
     def __init__(self, job, stat_spread):
+        """
+        :param job: Job enum, tells the character's job
+        :param stat_spread: dataclass containing all the attributes
+        """
         self.job = job
         self.wd = stat_spread.wd
-        self.stats = {}
-        self.stats[Stats.MAINSTAT] = Stat(Stats.MAINSTAT, stat_spread.mainstat)
-        self.stats[Stats.DET] = Stat(Stats.DET, stat_spread.det)
-        self.stats[Stats.CRIT] = ProbabalisticStat(Stats.CRIT, stat_spread.crit)
-        self.stats[Stats.DH] =  ProbabalisticStat(Stats.DH, stat_spread.dh)
-        self.stats[Stats.SPEED] = Stat(Stats.SPEED, stat_spread.speed)
-        self.stats[Stats.TEN] = Stat(Stats.TEN, stat_spread.ten)
-        self.stats[Stats.PIE] = Stat(Stats.PIE, stat_spread.pie)
+        self.character_stats = {
+            Stats.MAINSTAT: Stat(Stats.MAINSTAT, stat_spread.mainstat),
+            Stats.DET: Stat(Stats.DET, stat_spread.det),
+            Stats.CRIT: ProbabalisticStat(Stats.CRIT, stat_spread.crit),
+            Stats.DH: ProbabalisticStat(Stats.DH, stat_spread.dh),
+            Stats.SPEED: Stat(Stats.SPEED, stat_spread.speed),
+            Stats.TEN: Stat(Stats.TEN, stat_spread.ten),
+            Stats.PIE: Stat(Stats.PIE, stat_spread.pie),
+        }
 
     def get_gcd(self):
         """
         Returns the character's gcd given its skill or spell speed
         :returns: the gcd in seconds
         """
-        return math.floor(0.25 * (1000 - self.stats[Stats.SPEED].get_multiplier())) / 100
+        return math.floor(0.25 * (1000 - self.character_stats[Stats.SPEED].get_multiplier())) / 100
 
     def get_dot_scalar(self):
         """
         Returns the character's dot damage bonus, given its skill or spell speed
         :returns: the modified dot multiplier
         """
-        return  1 + (self.stats[Stats.SPEED].get_multiplier() / 1000)
+        return 1 + (self.character_stats[Stats.SPEED].get_multiplier() / 1000)
 
     def calc_piety(self):
         """
         Returns the character's mp regen resulting from the piety stat
         :returns: The character's mp regen per tick (3 seconds)
         """
-        return 200 + self.stats[Stats.PIE].get_multiplier()
+        return 200 + self.character_stats[Stats.PIE].get_multiplier()
 
-    # todo: break this up neatly
-    def calc_damage(self, potency, comp, is_dot=False, crit_rate=None, dh_rate=None):  #pylint: disable=too-many-arguments, too-many-locals
+    def calc_non_probablistic_damage(self, potency_per_second: float, comp: Comp = None, is_dot: bool = False) -> float:
```

Comment:
Small typing nit here: I'd use `Optional[Comp]` for the comp type, since you're explicitly setting it to None and actually checking for a None value later.

Reply:
fixed